src/plugins/validation/2and3/semantic-validators/operation-ids.js

```diff
@@ -57,47 +59,87 @@ module.exports.validate = function({ resolvedSpec }) {
   const operationIdPassedConventionCheck = (
     opKey,
     operationId,
-    hasPathParam
+    allPathOperations,
+    pathEndsWithParam
   ) => {
     // Only consider paths for which
-    // - paths with no path param has a GET and POST path
-    // - paths with path param has a GET, a DELETE, and a POST or PUT or PATCH.
+    // - paths that do not end with param has a GET and POST operation
+    // - paths that end with param has a GET, DELETE, POST, PUT or PATCH.
 
     let checkPassed = true;
+    const verbs = [];
 
-    if (!hasPathParam) {
+    if (!pathEndsWithParam) {
       // operationId for GET should starts with "list"
       if (opKey === 'get' && !operationId.match(/^list[a-zA-Z0-9_]+/m)) {
         checkPassed = false;
+        verbs.push('list');
       }
 
       // operationId for POST should starts with "create" or "add"
-      if (
+      else if (
         opKey === 'post' &&
         !operationId.match(/^(add|create)[a-zA-Z0-9_]+/m)
       ) {
         checkPassed = false;
+        verbs.push('add');
+        verbs.push('create');
       }
     } else {
       // operationId for GET should starts with "get"
       if (opKey === 'get' && !operationId.match(/^get[a-zA-Z0-9_]+/m)) {
         checkPassed = false;
+        verbs.push('get');
       }
 
       // operationId for DELETE should starts with "delete"
-      if (opKey === 'delete' && !operationId.match(/^delete[a-zA-Z0-9_]+/m)) {
+      else if (
+        opKey === 'delete' &&
+        !operationId.match(/^delete[a-zA-Z0-9_]+/m)
+      ) {
         checkPassed = false;
+        verbs.push('delete');
       }
 
-      // operationId for POST/PUT/PATCH should starts with "update"
-      if (
-        ['put', 'post', 'patch'].includes(opKey) &&
+      // operationId for PATCH should starts with "update"
+      else if (
+        opKey === 'patch' &&
         !operationId.match(/^update[a-zA-Z0-9_]+/m)
       ) {
         checkPassed = false;
+        verbs.push('update');
+      } else if (opKey === 'post') {
+        // operationId for POST should start with "add" or "create" only if PATCH operation exist for this path
+        if (
+          allPathOperations.includes('patch') &&
+          !operationId.match(/^(add|create)[a-zA-Z0-9_]+/m)
+        ) {
+          checkPassed = false;
+          verbs.push('add');
+          verbs.push('create');
+
+          // If PATCH operation doesn't exist for path, POST operationId should start with "update", "add" or "update"
```

Comment:
If there is no `PATCH` operation, then we should only allow "update" for `POST`.